lib/class-wp-theme-json-gutenberg.php

```diff
@@ -1824,7 +1825,7 @@ protected static function get_settings_values_by_slug( $settings, $preset_metada
 					is_callable( $preset_metadata['value_func'] )
 				) {
 					$value_func = $preset_metadata['value_func'];
-					$value      = call_user_func( $value_func, $preset );
+					$value      = call_user_func( $value_func, $preset, $settings );
```

Comment:
Not sure yet, but this might also provide a path towards CSS var resolution for layout values. So, 


```json
	"settings": {
		"appearanceTools": true,
		"layout": {
			"contentSize": ""var(--wp--custom--layout--content-size)",
			"wideSize": "1100px"
		},
		"typography": {
			"fluid": {
				"maxViewportWidth": {
					"ref": "settings.layout.contentSize"
				},
				"minViewportWidth": "600px"
			}
		}
	},
```

See:

- https://github.com/WordPress/gutenberg/issues/53525
- https://github.com/WordPress/gutenberg/issues/55070